### ISSUE TITLE
Mutant survives wrongly on a .NET 4.8 solution

### ISSUE THREAD

**Issue by EelFigure** [stryker-mutator/stryker-net#2245]
Hi,

yesterday I started working with Stryker.NET, the most recent version. It’s child’s play. In 15 minutes (including reading Quickstart) one has the local environment up and running. Very nice ! 

I tested with both a full framework 4.8 solution and a .NET6 solution. I had recently done this pluralsight course (https://app.pluralsight.com/library/courses/clean-architecture-patterns-practices-principles/exercise-files) and I was able to use the example solutions with reasonable test coverage from this course for the test.

**In case of the 4.8 variant, as far as I'm concerned, 1 mutant is wrongly not killed.** See the first attachment. In the .NET6 solution (which has the same logic code base and the same unit tests) this mutant is killed! See the 2nd attachment.
Is this a bug? I re-enacted it by manually making the change myself in the source code of both the .NET 4.8 and .NET6 variants and in both cases the unit test failed. With the .NET4.8 variant I therefore expected that the mutant would not survive.
Apparently there is a difference in implementation between what Stryker.NET does for 4.8 compared to NET6...

 - OS: Windows 10
 - Type of project: framework and Core
 - Framework Version: .Net full framework 4.8 and .NET6
 - Stryker Version 3.0.1

![attachment1](https://user-images.githubusercontent.com/25712980/196054543-3c5f2e8d-8a85-418d-a5b5-583364b1023a.png)
![attachment2](https://user-images.githubusercontent.com/25712980/196054545-c7974654-2cf2-450b-9512-a98d090f4bea.png)

**Comment by rouke-broersma** [comment]
Could you retest with coverage analysis off? https://stryker-mutator.io/docs/stryker-net/configuration/#coverage-analysis-string

**Comment by EelFigure** [comment]
Unfortunately the retest with coverage analysis off in the stryker-config file does not work.

![Stryker-config](https://user-images.githubusercontent.com/25712980/196239863-7e776685-d583-445c-beab-ce8803658e32.PNG)

**Comment by EelFigure** [comment]
The code is behind a login on pluralsight: https://app.pluralsight.com/library/courses/clean-architecture-patterns-practices-principles/exercise-files. The problem exists within the .NET 4.8 full framework solution. The 4.8 solution can also be found here: https://github.com/matthewrenze/clean-architecture-demo. The .NET6 solution can also be found here: https://github.com/matthewrenze/clean-architecture-core

**Comment by dupdob** [comment]
Thanks for the reference. I downloaded the source to try and reproduce the issue. I am working with the Net 4.8 version for now.
I figured out that tests and production code share the same assembly. **This scenario is not supported by Stryker.Net as of today**. As a matter of fact, Stryker.Net mutates the tests as well as production code in the application project.

So, can you please specify which project (or folder) you choose for testing ?

BTW, I noticed in your snapshots that the 'issue' concerns every mutant in this method.

As a general rules, Stryker has limitations when running on full framework projects vs .net ones; I suspect you may run into one of those limitations

**Comment by EelFigure** [comment]
I can understand why Stryker.Net does not support tests and production code in the same assembly. Unit tests should not be shipped to production environment, that would make no sense. 

My goal is to have the code in the Application.csproj getting mutated. The 6 tests in the specification.csproj were run in the initial test run. See the screenprint.

![CMD_Stryker_NET](https://user-images.githubusercontent.com/25712980/197347572-5ba84c8a-21f6-4cf0-8964-b19d123ae915.PNG)

**Comment by dupdob** [comment]
Thanks for the details, I actually did the same thing on the Net 4.8 project. As far as I can tell, this is normal for this mutation to be a survivor: manually running the tests showed that _Specification_'s were all green: all of them passed.
Using another coverage analysis tool that no _Specification_'s test actually cover the `CreateSaleCommand` class.
So, if there is an issue, it is with Net 6. I will see what happens

**Comment by dupdob** [comment]
Well, this is awkward: there is also no problem with Stryker and the Net 6 version of this project: when manually injecting this mutation, one one _Specification_'s test fails. So this is normal for Stryker to flag it as killed.
The question is: why is there a behavior difference on the project side. This relates either to subtle differences between the two projects, or implementation details in Specflow (which I am familiar with, but I have not used it in years).
I am afraid I do not have time to dig further into this.

Bottom line is: this is not a Stryker issue.
